src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleKeyCache.scala

```diff
@@ -123,15 +127,20 @@ class GoogleKeyCache(
   private def retrieveActiveKey(pet: PetServiceAccount): IO[(Option[String], List[GcsObjectName], List[ServiceAccountKey])] =
     for {
       (keysFromCache, keysFromIam) <- fetchKeysFromCacheAndIam(pet)
-      maybeMostRecentKey = keysFromCache.sortBy(_.timeCreated.toEpochMilli).lastOption
-      mostRecentKey <- maybeMostRecentKey match {
-        case Some(mostRecentKey) if isKeyActive(mostRecentKey, keysFromIam) =>
+
+      // TODO CORE-278: this could result in log spam
+      _ = if (keysFromIam.length >= 10)
+        logger.warn(s"danger: pet ${pet.serviceAccount.displayName.value} has ${keysFromIam.length} keys (cache has ${keysFromCache.length})")
+
+      maybeActiveKey = keysFromCache.sortBy(_.timeCreated.toEpochMilli).findLast(x => isKeyActive(x, keysFromIam))
```

Comment:
in cases where the bucket-based key cache and the actual IAM list of keys are out of sync for whatever reason, this change will be slightly more flexible in finding a usable key. It will only make a difference in cases where the pet has multiple keys already. If there is some kind of race condition or contention where we create multiple keys in rapid succession, this change should have a better chance at finding a usable one.